lib/adagrams.rb

```diff
@@ -0,0 +1,134 @@
+# Sources: https://stackoverflow.com/questions/33160746/is-there-anything-bsearch-for-strings-in-ruby
+
+require 'csv'
+
+ALL_LETTERS = {
+    A: 9,
+    B: 2,
+    C: 2,
+    D: 4,
+    E: 12,
+    F: 2,
+    G: 3,
+    H: 2,
+    I: 9,
+    J: 1,
+    K: 1,
+    L: 4,
+    M: 2,
+    N: 6,
+    O: 8,
+    P: 2,
+    Q: 1,
+    R: 6,
+    S: 4,
+    T: 6,
+    U: 4,
+    V: 2,
+    W: 2,
+    X: 1,
+    Y: 2,
+    Z: 1
+}
+
+SCORE_CHART = {
+    A: 1,
+    B: 3,
+    C: 3,
+    D: 2,
+    E: 1,
+    F: 4,
+    G: 2,
+    H: 4,
+    I: 1,
+    J: 8,
+    K: 5,
+    L: 1,
+    M: 3,
+    N: 1,
+    O: 1,
+    P: 3,
+    Q: 10,
+    R: 1,
+    S: 1,
+    T: 1,
+    U: 1,
+    V: 4,
+    W: 4,
+    X: 8,
+    Y: 4,
+    Z: 10
+}
+
+def draw_letters
+  all_letters = ALL_LETTERS.map do |keys, values|
```

Comment:
This only operates on a _single_ key and value at a time.

```suggestion
  all_letters = ALL_LETTERS.map do |key, value|
```